packages/databyss-ui/components/Editor/state/reducer.js

```diff
@@ -33,11 +34,15 @@ const cleanUpState = state => {
     if (type === 'ENTRY') {
       _entries[refId] = entries[refId]
     }
+    if (type === 'TOPIC') {
```

Comment:
We're also repeating ourselves a lot when we are accessing entities in the state. How about a selector like
```
const entities = (state, type) => {
  ENTRY: state.entries,
  SOURCE: state.sources,
  TOPIC: state.topics,
  AUTHOR: state.authors,
}[type]
```

Then we can write this as:
```
pageBlocks.forEach(b => {
  const { type, refId } = blocks[b._id]
  entities(state, type)[refId] = entities(state, type)[refId]
}
```